src/mon/MgrMonitor.cc

```diff
@@ -955,6 +989,11 @@ bool MgrMonitor::prepare_command(MonOpRequestRef op)
       r = -EINVAL;
       goto out;
     }
+    if (pending_map.always_on_modules.find(module) !=
+        pending_map.always_on_modules.end()) {
+      ss << "module '" << module << "' cannot be disabled (always-on)";
+      goto out;
```

Comment:
might want to set `r` to `-EINVAL`, so we can reply the user right away.

Reply:
shall we update the return codes for other situations? currently the user gets a success/0 return code when (1) enabling an already enabled module, (2) disabling an already disabled module, and (3) disabling a non-existent module. The first two seem OK, but (3) is similar to this case. My reasoning behind this semantics is that a user script would receive an error for functionality it can't change anyway...